pkg/services/pluginsintegration/clientmiddleware/tracing_header_middleware.go

```diff
@@ -50,6 +52,7 @@ func (m *TracingHeaderMiddleware) applyHeaders(ctx context.Context, req backend.
 		if gotVal == "" {
 			continue
 		}
+		gotVal = sanitizeGrpcHeaderValue(gotVal)
```

Comment:
this is just me being extra careful but since I don't know how expensensive is this to calculate and it's only necessary for human input, I would sanitize only the new headers (the title ones). We have other guards in other places so things like a `DatasourceUID` shouldn't have unexpected characters.

Reply:
You could make use of `utf8.Valid(gotVal)` to avoid having to do it all the time for example. That would also signal more clearly this code is about preventing UTF-8 encoding/decoding issues